insecure/corruptlibp2p/pubsubAdapter.go

```diff
@@ -0,0 +1,120 @@
+package corruptlibp2p
+
+import (
+	"context"
+	"fmt"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/rs/zerolog"
+	corrupt "github.com/yhassanzadeh13/go-libp2p-pubsub"
+
+	"github.com/onflow/flow-go/insecure/internal"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/utils/logging"
+)
+
+// CorruptGossipSubAdapter is a wrapper around the forked pubsub topic from
+// github.com/yhassanzadeh13/go-libp2p-pubsub that implements the p2p.PubSubAdapter.
+// This is needed because in order to use the forked pubsub module, we need to
+// use the entire dependency tree of the forked module which is resolved to
+// github.com/yhassanzadeh13/go-libp2p-pubsub. This means that we cannot use
+// the original libp2p pubsub module in the same package.
+// Note: we use the forked pubsub module for sake of BFT testing and attack vector
+// implementation, it is designed to be completely isolated in the "insecure" package, and
+// totally separated from the rest of the codebase.
+type CorruptGossipSubAdapter struct {
+	gossipSub *corrupt.PubSub
+	router    *internal.CorruptGossipSubRouter
+	logger    zerolog.Logger
+}
+
+var _ p2p.PubSubAdapter = (*CorruptGossipSubAdapter)(nil)
+
+func (c *CorruptGossipSubAdapter) RegisterTopicValidator(topic string, topicValidator p2p.TopicValidatorFunc) error {
+	// instantiates a corrupt.ValidatorEx that wraps the topicValidatorFunc
+	var corruptValidator corrupt.ValidatorEx = func(ctx context.Context, from peer.ID, message *corrupt.Message) corrupt.ValidationResult {
+		pubsubMsg := &pubsub.Message{
+			Message:       message.Message, // converting corrupt.Message to pubsub.Message
+			ID:            message.ID,
+			ReceivedFrom:  message.ReceivedFrom,
+			ValidatorData: message.ValidatorData,
+			Local:         message.Local,
+		}
+		result := topicValidator(ctx, from, pubsubMsg)
+
+		// overriding the corrupt.ValidationResult with the result from pubsub.TopicValidatorFunc
+		message.ValidatorData = pubsubMsg.ValidatorData
+
+		switch result {
+		case p2p.ValidationAccept:
+			return corrupt.ValidationAccept
+		case p2p.ValidationIgnore:
+			return corrupt.ValidationIgnore
+		case p2p.ValidationReject:
+			return corrupt.ValidationReject
+		default:
+			// should never happen, indicates a bug in the topic validator
+			c.logger.Fatal().Msgf("invalid validation result: %v", result)
+		}
+		// should never happen, indicates a bug in the topic validator, but we need to return something
+		c.logger.Warn().
```

Comment:
Same comment as line 59 add some message context to the log.

Reply:
https://github.com/onflow/flow-go/pull/3585/commits/2cbea2a4f0df1643b72ea1b2c1c8ed4d9ec030e7